packages/convex-helpers/server/pagination.ts

```diff
@@ -314,3 +318,254 @@ function getIndexKey<
   }
   return key;
 }
+
+const END_CURSOR = "endcursor";
+
+/**
+ * Simpified version of `getPage` that you can use for one-off queries that
+ * don't need to be reactive.
+ * 
+ * These two queries are roughly equivalent:
+ * 
+ * ```ts
+ * await ctx.db.query(table)
+ *  .withIndex(index, q=>q.eq(field, value))
+ *  .filter(q=>q.neq(q.field(field1), value1))
+ *  .order("desc")
+ *  .paginate(opts)
+ *
+ * await getPageOfQuery(
+ *   ctx,
+ *   db=>db.query(table).withIndex(index, q=>q.eq(field, value)).order("desc"),
+ *   opts,
+ *   {
+ *     filter: async (doc) => doc[field1] !== value1,
+ *     schema,
+ *   },
+ * )
+ * ```
+ * 
+ * Differences:
+ * 
+ * - getPageOfQuery does not automatically pin the end of the page for when
+ *   the query reruns. To pin the end of the page, you can use the `endCursor`
+ *   option. This does not happen automatically.
+ * - getPageOfQuery can be called multiple times in a query or mutation,
+ *   and within Convex components.
+ * - Cursors are not encrypted.
+ * 
+ * @argument paginationOpts.cursor Where to start the page. This should come from
+ * `continueCursor` in the previous page.
+ * @argument paginationOpts.endCursor Where to end the page. This should from from
+ * `continueCursor` in the *current* page.
+ * NOTE: `endCursor` is not automatically populated. If you want reactive
+ * pagination, a la
+ * [this article](https://stack.convex.dev/fully-reactive-pagination),
+ * you need to rerun the query with `endCursor` provided.
+ * If not provided, the page will end when it reaches `options.opts.numItems`.
+ * @argument options.filter The filter is optional, and it's a post-filter
+ * like the convex-helpers `filter()`, so a sparse predicate may result
+ * in small or empty pages.
```

Comment:
```suggestion
 * in small or empty pages, though a best effort is made to retrieve `numItems`.
```

Reply:
i'm not sure what "best effort" means there. we look at `numItems` documents and then apply the filter

Reply:
oh interesting. I thought we tried with some limit to fill `numItems`. If we're truly just fetching `numItems` then why offer `filter` at all?

Reply:
as a convenience to make it easy to switch from the native syntax, and because the `filter()` convex-helper doesn't compose with this helper

Reply:
I had assumed it would work harder to ensure it returned `numItems` so I'd rather snip it and just suggest folks do
```ts
(await getPageOfQuery(ctx, ...)).filter(doc => doc.isGood)
```
This also means that the type of `paginator(ctx, schema, tableName)` could just be the same as `db.query` (and just throw at runtime for unimplemented functions) since all the syntax would conform as a subset.